common/src/main/java/com/sequenceiq/cloudbreak/quartz/saltstatuschecker/SaltStatusCheckerJobService.java

```diff
@@ -0,0 +1,97 @@
+package com.sequenceiq.cloudbreak.quartz.saltstatuschecker;
+
+import java.security.SecureRandom;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.quartz.JobBuilder;
+import org.quartz.JobDetail;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.cloudbreak.quartz.JobDataMapProvider;
+import com.sequenceiq.cloudbreak.quartz.model.JobResourceAdapter;
+
+public abstract class SaltStatusCheckerJobService<T extends JobResourceAdapter<?>> {
```

Comment:
None of the *JobService classes have unit tests, this is why I have not created it. I think it would be best to extract the common part of these classes to ease the testing of them

Reply:
then please at least file some jira to improve this. Not having tests for the other cases doesn't mean we should never write test for these

Reply:
Sure, I have created it: https://jira.cloudera.com/browse/CB-17941